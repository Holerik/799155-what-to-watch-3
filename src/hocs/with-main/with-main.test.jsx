// with-main.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import MainPage from '../../components/main/main.jsx';
import withMain from './with-main.jsx';


const Main = withMain(MainPage);

const filmsInfo = [
  {
    id: 555,
    title: `With Unsane`,
    poster: `img/with-unsane.jpg`,
    altPoster: `whis-unsane poster`,
    src: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
    preview: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
  },
  {
    id: 255,
    title: `Second With Unsane`,
    poster: `img/second-with-unsane.jpg`,
    altPoster: `whis-second-unsane poster`,
    src: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
    preview: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
  }
];

const filmsFullInfo = [
  {
    id: 2357,
    title: `With Little Women`,
    poster: `img/with-little-woman-poster.jpg`,
    altPoster: `with-little Women poster`,
    background: `img/with-little-wooman-bg.jpg`,
    altBackground: `Movie background`,
    description: `Based on Loiusa May Alcott’s classic novel with the same name, Little Women is a timeless tale of love, life and ambitions`,
    review: `A writer, a sketch artist, an aspiring actress and a budding pianist. All four March sisters are gifted with a unique talent, but they live in the times where the only two choices for women are marriage and death. Writer-director Greta Gerwig`,
    reviews: [0, 1, 2, 3, 4],
    genre: [`Drama`, `Romance`],
    year: 2019,
    duration: `2h 15min`,
    age: `16+`,
    rating: {
      score: `8.0`,
      level: `very good`,
      count: 150,
    },
    director: `Greta Gerwig`,
    starring: [`Saoirse Ronan`, `Emma Watson`, `Florence Pugh`, `Meryl Streep`, `Timothee Chalamet`],
    src: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
    preview: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
  }
];

const promoMovie = {
  id: 150,
  title: `With Fantomas`,
  poster: `img/with-fantomas-poster.jpg`,
  altPoster: `With Fantomas poster`,
  background: `img/with-fantomas-bg.jpg`,
  altBackground: `With Fantomas background`,
  description: `The best men of France - a brave journalist and an extremely energetic commissioner - attack the trail of a mysterious criminal mastermind`,
  review: `A supervillain known as Fantomas tries to foil a couple of foes: the police inspector who is trying to catch him and the reporter who doesn’t believe in his existence`,
  reviews: [0, 1, 2, 3, 4],
  genre: [`Comedy`, `Adventure`, `Crime`],
  year: 1964,
  duration: `1h 40min`,
  age: `16+`,
  rating: {
    score: `7.0`,
    level: `very good`,
    count: 110,
  },
  director: `André Hunebelle`,
  starring: [`Jean Marais`, `Louis de Funès`, `Mylène Demongeot`, `Jacques Dynam`, `Robert Dalban`],
  src: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
  preview: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
};

const setMovieCardId = () => {};
const setPageId = () => {};
const setGenre = () => {};
const playButtonClickHandler = () => {};
const listButtonClickHandler = () => {};

it(`Should correctly render the main page`, () => {
  const tree = renderer.create(
      <Main
        filmsInfo={filmsInfo}
        filmsFullInfo={filmsFullInfo}
        setMovieId={setMovieCardId}
        promoMovie={promoMovie}
        setPageId={setPageId}
        setGenre={setGenre}
        genre={`All genres`}
        playButtonClickHandler={playButtonClickHandler}
        listButtonClickHandler={listButtonClickHandler}
      >
      </Main>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

