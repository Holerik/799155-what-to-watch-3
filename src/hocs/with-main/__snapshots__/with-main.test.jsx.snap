// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render the main page 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="With Fantomas background"
        src="img/with-fantomas-bg.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="With Fantomas poster"
            height="327"
            src="img/with-fantomas-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            With Fantomas
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Comedy·Adventure·Crime
            </span>
            <span
              className="movie-card__year"
            >
              1964
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <ul
      className="catalog__genres-list"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
       
      <li
        className="catalog__genres-item
                      catalog__genres-item--active"
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          All genres
        </a>
      </li>
      <li
        className="catalog__genres-item
                      "
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Drama
        </a>
      </li>
      <li
        className="catalog__genres-item
                      "
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Romance
        </a>
      </li>
    </ul>
    <section
      className="catalog"
    >
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            id="555"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              alt="whis-unsane poster"
              height="175"
              src="img/with-unsane.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="#"
            >
              With Unsane
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            id="255"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              alt="whis-second-unsane poster"
              height="175"
              src="img/second-with-unsane.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="#"
            >
              Second With Unsane
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button visually-hidden"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
